src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Symbols/SymbolLoader.cs

```diff
@@ -304,7 +302,7 @@ private bool HasArrayConversionToInterface(ArrayType pSource, CType pDest)
                 return true;
             }
 
-            AggregateType atsDest = pDest.AsAggregateType();
+            AggregateType atsDest = pDest as AggregateType;
```

Comment:
Or better again IMO, direct cast and then use it in the next line. It's nicer to call something on `AggregateType` than on its base that then asserts `this is AggregateType`.